packages/core-utils/src/fp-utils.ts

```diff
@@ -0,0 +1,22 @@
+import _ from 'lodash/fp';
+
+type AppendFn = {
+  <T>(value: T): (list: T[]) => T[];
+  <T>(value: T, list: T[]): T[];
+};
```

Comment:
It appeared that the type parameter didn't carry through `safeCurry`, so added a type with both signatures here.

Reply:
FYI, safeCurry wasn't used...

`export const append = _.curry((value, arr) => _.concat(arr, [value]));`

safeCurry should do the same function overload type flow you are establishing here...  unless you tried that and it didn't work?

Reply:
Yeah, `safeCurry` didn't carry through the type parameter.

Reply:
what if you included the return type in the inner signature?  lodash functions like to any-sledgehammer types

```
export const append = safeCurry(<T>(value: T, arr: T[]): T[] => _.concat(arr, [value]));
const appendFoo = append('foo');
```

Reply:
With that, `appendFoo`'s type is `any`.

Reply:
Same result.

Reply:
uggg.... its Friday and I'm being forced to think about how to better type-flow curried functions... ARG!!!!

Reply:
Is there an issue with the `AppenFn` type added here? It seems fine to me if typing generic curried is a little verbose, especially if we want to limit curried functions.